modules/dnn/src/onnx/onnx_importer.cpp

```diff
@@ -2820,6 +2820,10 @@ void ONNXImporter::parseElementWise(LayerParams& layerParams, const opencv_onnx:
         std::vector<Mat> inputs, output;
         for (size_t i = 0; i < node_proto.input_size(); ++i)
         {
+            auto input_i = getBlob(node_proto, i);
+            if (i == 1) {
+                input_i.convertTo(input_i, CV_32F);
+            }
```

Comment:
A bit confusing. Why `if (i == 1` but not `if (input_i.type() != CV_32F)?

Also, what happens with input_i after conversion?

Reply:
I mean it's more clear to do like this:
```cpp
for (size_t i = 0; i < node_proto.input_size(); ++i)
{
    auto input = getBlob(node_proto, i);
    if (input.type() != CV_32F)
        input.convertTo(input, CV_32F); 
    inputs.push_back(input);
}
```

Reply:
There is `Where` operator which is parsed by this method as well. I did this just to avoid some potential issues, but still I can try your solution.